xtask/src/main.rs

```diff
@@ -0,0 +1,284 @@
+}
+
+#[derive(Debug, Args)]
+struct GenerateEfuseFieldsArgs {
+    /// Local path to the `esptool` repository
+    esptool_path: PathBuf,
+}
+
+// ----------------------------------------------------------------------------
+// Application
+
+fn main() -> Result<()> {
+    env_logger::Builder::new()
+        .filter_module("xtask", log::LevelFilter::Info)
+        .init();
+
+    // The directory containing the cargo manifest for the 'xtask' package is a
+    // subdirectory within the cargo workspace:
+    let workspace = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    let workspace = workspace.parent().unwrap().canonicalize()?;
+
+    match Cli::parse() {
+        Cli::GenerateEfuseFields(args) => generate_efuse_fields(&workspace, args),
+    }
+}
+
+// ----------------------------------------------------------------------------
+// Generate eFuse Fields
+
+const HEADER: &str = r#"
+//! This file was automatically generated, please do not edit it manually!
+//! 
+//! Generated: $DATE
+//! Version:   $VERSION
+
+#![allow(unused)]
+
+use super::EfuseField;
+
+"#;
+
+type EfuseFields = HashMap<String, EfuseYaml>;
+
+#[derive(Debug, serde::Deserialize)]
+struct EfuseYaml {
+    #[serde(rename = "VER_NO")]
+    version: String,
+    #[serde(rename = "EFUSES")]
+    fields: HashMap<String, EfuseAttrs>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Deserialize)]
+struct EfuseAttrs {
+    #[serde(rename = "blk")]
+    block: u32,
+    word: u32,
+    len: u32,
+    start: u32,
+    #[serde(rename = "desc")]
+    description: String,
+}
+
+impl PartialOrd for EfuseAttrs {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for EfuseAttrs {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.block.cmp(&other.block) {
+            Ordering::Equal => {}
+            ord => return ord,
+        }
+
+        match self.word.cmp(&other.word) {
+            Ordering::Equal => {}
+            ord => return ord,
+        }
+
+        self.start.cmp(&other.start)
+    }
+}
+
+fn generate_efuse_fields(workspace: &Path, args: GenerateEfuseFieldsArgs) -> Result<()> {
+    let efuse_yaml_path = args
+        .esptool_path
+        .join("espefuse")
+        .join("efuse_defs")
+        .canonicalize()?;
+
+    let espflash_path = workspace.join("espflash").canonicalize()?;
+
+    let mut efuse_fields = parse_efuse_fields(&efuse_yaml_path)?;
+    process_efuse_definitions(&mut efuse_fields)?;
+    generate_efuse_definitions(&espflash_path, efuse_fields)?;
+
+    Command::new("cargo")
+        .args(["+nightly", "fmt"])
+        .current_dir(workspace)
+        .output()?;
+
+    Ok(())
+}
+
+fn parse_efuse_fields(efuse_yaml_path: &Path) -> Result<EfuseFields> {
+    // TODO: We can probably handle this better, e.g. by defining a `Chip` enum
+    //       which can be iterated over, but for now this is good enough.
+    const CHIPS: &[&str] = &[
```

Comment:
You can depend on esp-metadata and get this for free

Reply:
`esp-metadata` doesn't have the P4 (solvable of course), and pulls in some additional dependencies, so I just didn't bother. I think given that we probably won't need to run the subcommand until we add support for a new chip, which who knows when that will be, it's okay as-is for the time being.